store/postgres/src/relational_queries.rs

```diff
@@ -1406,6 +1406,15 @@ impl<'a> ParentLimit<'a> {
         }
         Ok(())
     }
+
+    /// Include a 'limit {num_parents}+1' clause for single-object queries
+    /// if that is needed
+    fn single_limit(&self, num_parents: usize, out: &mut AstPass<Pg>) {
+        if let ParentLimit::Ranked(_, _) = self {
```

Comment:
Could this do a match on `ParentLimit`, for future-proofing, and give an explanation of why the `Outer` variant doesn't need this?

Reply:
Good point. Added a commit that does that